lib/ansible/modules/cloud/kubevirt/kubevirt_scale_vmirs.py

```diff
@@ -132,117 +88,169 @@
-                self.exit_json(changed=False)
-
-            api_response = api_instance.patch_namespaced_custom_object(
-                group, version, namespace, plural, name, body)
-            self.exit_json(changed=True, result=api_response)
-        except ApiException as exc:
-            self.fail_json(msg='Failed to manage requested object',
-                           error=exc.reason)
-
-    def __authenticate(self):
-        """ Build API client based on user's configuration """
-        host = self.params.get('host')
-        username = self.params.get('username')
-        password = self.params.get('password')
-        api_key = self.params.get('api_key')
-
-        if (host and username and password) or (api_key and host):
-            return self.__configure_by_params()
-        return self.__configure_by_file()
-
-    def __configure_by_file(self):
-        if not self.params.get('kubeconfig'):
-            config_file = os.path.expanduser(
-                kube_config.KUBE_CONFIG_DEFAULT_LOCATION)
-        else:
-            config_file = self.params.get('kubeconfig')
+        scale_obj = {'metadata': {'name': name, 'namespace': namespace}, 'spec': {'replicas': replicas}}
+
+        return_obj = None
+        stream = None
+
+        if wait:
+            w, stream = self._create_stream(resource, namespace, wait_time)
 
         try:
-            kube_config.load_kube_config(
-                config_file=config_file,
-                context=self.params.get('context')
-            )
-            config = kubernetes.client.Configuration()
-            if not self.params.get('verify_ssl'):
-                setattr(config, 'verify_ssl', False)
-            return kubernetes.client.ApiClient(
-                configuration=config
+            self.patch_resource(resource, scale_obj, existing_object, name,
+                                namespace, merge_type='merge')
+        except Exception as exc:
+            self.fail_json(
+                msg="Scale request failed: {0}".format(exc.message)
             )
-        except (IOError, ConfigException):
-            raise
-
-    def __configure_by_params(self):
-        kube_config.load_kube_config()
-        config = kubernetes.client.Configuration()
-        auth_args = AUTH_ARG_SPEC.keys()
-
-        for key, value in iteritems(self.params):
-            if key in auth_args and value is not None:
-                if key == 'api_key':
-                    setattr(
-                        config,
-                        key, {'authorization': "Bearer {0}".format(value)})
-                else:
-                    setattr(config, key, value)
-
-        if not self.params.get('verify_ssl'):
-            setattr(config, 'verify_ssl', False)
-
-        return kubernetes.client.ApiClient(configuration=config)
+
+        if wait and stream is not None:
+            return_obj = self._read_stream(resource, w, stream, name, replicas)
+
+        if not return_obj:
```

Comment:
I still do not understand this if. Let's say that `wait` is `None` and we won't read stream. As a result `return_obj` will be None so we will ?? wait. My question is why should we if said we do not want to by not setting `wait`.

Reply:
I've removed it, we now return the object returned by `patch_resource` method.